lib/chef/event_dispatch/dispatcher.rb

```diff
@@ -28,6 +28,9 @@ def register(subscriber)
       # Define a method that will be forwarded to all
       def self.def_forwarding_method(method_name)
         define_method(method_name) do |*args|
+          if method_name == :deprecation && args.size == 1
```

Comment:
@coderanger that won't make a difference though, the issue is we need to know who *originally* called `deprecation` and if we wait until after dispatcher invokes the "real" `deprecation`, the caller will just be the dispatcher.